### PR TITLE
add gang\gpu\quota

### PR DESCRIPTION
**Pull request by buptcozy**
Signed-off-by: xingbao.zy <xingbao.zy@alibaba-inc.com>

### CODE REVIEW

File: docs/user-manuals/multi-hierarchy-elastic-quota-management.md [hunk 1]
```diff
@@ -0,0 +1,519 @@
+# Multi-Hierarchy-ElasticQuota-Management
+
+Multi-Hierarchy-ElasticQuota-Management is an ability of koord-scheduler to manage different user's resource usage in a shared-cluster.
+
+## Introduction
+When several users or teams share a cluster, fairness of resource allocation is very important. This proposal provides
+multi-hierarchy elastic quota management mechanism for the scheduler.
```
Comment (hormes) [inline docs/user-manuals/multi-hierarchy-elastic-quota-management.md:7]: This proposal provides
multi-hierarchy elastic quota management mechanism for the scheduler.
-->
Koordinator provides
multi-hierarchy elastic quota management mechanism in the scheduler.

File: docs/user-manuals/multi-hierarchy-elastic-quota-management.md [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+
+3.Verify `quota-example` has changed.
+```bash
+$ kubectl get eqs -n default quota-example -o yaml
+```
+```yaml
+kind: ElasticQuota
+metadata:
+  annotations:
+    kubectl.kubernetes.io/last-applied-configuration: |
```
Comment (hormes) [inline docs/user-manuals/multi-hierarchy-elastic-quota-management.md:107]: remove the last-applied-configuration for better readability

File: docs/user-manuals/multi-hierarchy-elastic-quota-management.md [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+  min:
+    cpu: "10"
+    memory: 20Mi
+status:
+  used:
+    cpu: 40m
+    memory: 40Mi
+```
+
+4.Verify `quota-example` debug api.
```
Comment (hormes) [inline docs/user-manuals/multi-hierarchy-elastic-quota-management.md:138]: this step is optional, add a hint?

File: docs/user-manuals/multi-hierarchy-elastic-quota-management.md [hunk 4]
```diff
@@ -0,0 +1,519 @@
...
+```bash
+$ kubectl -n koordinator-system get lease koord-scheduler --no-headers | awk '{print $2}' | cut -d'_' -f1 | xargs -I {} kubectl -n koordinator-system get pod {} -o wide --no-headers | awk '{print $6}'
+$ curl $schedulerip::10251/apis/v1/plugins/ElasticQuota/quota/quota-example
```
Comment (hormes) [inline docs/user-manuals/multi-hierarchy-elastic-quota-management.md:141]: the `schedulerip` var is not assigned

File: docs/user-manuals/multi-hierarchy-elastic-quota-management.md [hunk 5]
```diff
@@ -0,0 +1,519 @@
...
+        "cpu": "40",
+        "memory": "40Gi",
+    },
+    "used": {
+        "cpu": "40m",
+        "memory": "40Mi"
+    }
+}
+```
+The main different with yaml is that we can find all quota's pods and its status in `podCache`.
```
Comment (hormes) [inline docs/user-manuals/multi-hierarchy-elastic-quota-management.md:190]: Consider the example of adding one or more namespace Pods to a Quota?

File: docs/user-manuals/multi-hierarchy-elastic-quota-management.md (hunk 1 above)
Comment (buptcozy) [reply docs/user-manuals/multi-hierarchy-elastic-quota-management.md:7]: fixed

File: docs/user-manuals/multi-hierarchy-elastic-quota-management.md (hunk 2 above)
Comment (buptcozy) [reply docs/user-manuals/multi-hierarchy-elastic-quota-management.md:107]: fixed

File: docs/user-manuals/multi-hierarchy-elastic-quota-management.md (hunk 3 above)
Comment (buptcozy) [reply docs/user-manuals/multi-hierarchy-elastic-quota-management.md:138]: fixed

File: docs/user-manuals/multi-hierarchy-elastic-quota-management.md (hunk 4 above)
Comment (buptcozy) [reply docs/user-manuals/multi-hierarchy-elastic-quota-management.md:141]: fixed

File: docs/user-manuals/multi-hierarchy-elastic-quota-management.md (hunk 5 above)
Comment (buptcozy) [reply docs/user-manuals/multi-hierarchy-elastic-quota-management.md:292]: fixed

**Pull request merged**